app/services/my_anime_list_sync_service.rb

```diff
@@ -15,32 +15,43 @@ def execute_method
 
     case method
     when 'delete'
-      delete("#{media_type}list/#{media_type}/#{mal_media_id}", linked_profile)
+      delete(
+        "#{media_type}list/#{media_type}/#{mal_media_id}",
+        linked_account
+      )
     when 'create/update'
       # find the anime or manga
       # it will raise an error if it fails the http request
-      response = get("#{media_type}/#{mal_media_id}#{MINE}", linked_profile)
+      response = get(
+        "#{media_type}/#{mal_media_id}#{MINE}",
+        linked_account
+      )
 
       if media_type == 'anime' && response['watched_status']
-        put("animelist/anime/#{mal_media_id}", linked_profile,
+        # anime already exists in their list
+        put("animelist/anime/#{mal_media_id}", linked_account,
           status: format_status(library_entry.status),
           episodes: library_entry.progress,
           score: format_score(library_entry.rating),
           rewatch_count: library_entry.reconsume_count)
       elsif media_type == 'anime'
-        post('animelist/anime', linked_profile,
+        # anime does not exist in their list
+        post('animelist/anime', linked_account,
           anime_id: mal_media_id,
           status: format_status(library_entry.status),
           episodes: library_entry.progress,
           score: format_score(library_entry.rating))
-      elsif media_type == 'manga' && response['read_status']
-        put("mangalist/manga/#{mal_media_id}", linked_profile,
+      elsif media_type == 'manga' &&
+            (response['id'].nil? || response['read_status'])
```

Comment:
pretty much the goal of it is, that once it gets fixed it will just continue to work.